### ISSUE TITLE
Keychain add/remove value in the background

### ISSUE THREAD

**Issue by lehelmedves** [kishikawakatsumi/KeychainAccess#523]
Hello,

This is an incredibly useful library, thank you for developing and maintaining it.
I have one problem, that I've run into and I'd like to pick your brain about it in hope of finding a solution to it.

I am trying to update a Keychain key-value pair right inside the following method, while receiving a silent push notification:
`func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void)`
It seems to work at first, however the new value doesn't really get saved, most probably because the app is only running for a short amount of time when woken up by the silent push notification. When I open the app and read back the value, it's still the old one.

Have you ever encountered this issue? Do you have any idea how this could be avoided?

Thank you in advance.

**Comment by yegorsch** [comment]
Are you calling the completion handler? The documentation says you should have up to 30s to process the notification.

**Comment by lehelmedves** [comment]
Yes, I do initiate the value update and right after that I call the completion handler.

**Comment by nattnh751** [comment]
I am having a similar problem to this.

**Comment by nattnh751** [comment]
https://github.com/evgenyneu/keychain-swift/issues/15  possibly another example of this issue.  Possible work around : https://github.com/evgenyneu/keychain-swift/issues/78#issuecomment-495959391.  Note I have not yet tried this!
